Rewrite RELEASE_LABEL_REGEX code to use .match, tweak regex

dobbymoodge opened the pull request:
These changes:

* fix a regex to extract version numbers from our `-fy18.q3` style
  labels for sorting
  * also support any non-digit prefix for the version/release number
    parts, e.g. `-fy18.q3.r2.4`
* drop some of the captured groups so that we only pull out the parts
  of the label that we're using
* update dependent code to use `.match` and to reference the capture
  groups in the result according to the changes mentioned above

Sometimes you need an ASCII diagram to understand a regex :/

dobbymoodge commented:
@sspeiche This addresses the sorting issue with card labels

danmcp commented:
@dobbymoodge You've changed this to the point I don't know what it's doing anymore and I am probably not going to have enough time to give it any justice any time soon.

dobbymoodge commented:
Fine, be that way @danmcp :)

@brenton can you give this a once-over? Basically I've:
* fixed a regex to extract version numbers from our `-fy18.q3` style labels for sorting
* dropped some of the captured groups so that we only pull out the parts of the label that we're using
* updated dependent code to use `.match` and reference the capture groups in the result according to the changes mentioned above

I should probably amend the commit message with this stuff...

stevekuznetsov commented on lib/trello_helper.rb:
```diff
@@ -28,8 +28,10 @@ class TrelloHelper
 
-  RELEASE_LABEL_REGEX = /^(proposed|targeted|committed)-((\w*)-)*((?:fy)?(\d+)(.((?:q)?\d+))?(.(\d+))?(.(\d+))*)/
-
+  RELEASE_LABEL_REGEX = /^(proposed|targeted|committed)-(?:(.+)-)?((?:\D*(\d+))\.(?:\D*(\d+))(?:\.(?:\D*(\d+)))?(?:\.(?:\D*(\d+)))?)/
```
:(

stevekuznetsov commented:
Unit tests?

stevekuznetsov commented:
Not useful for me to review that regex
